### DISCUSSION TITLE
Now for something completely different - Custom GPT for Labs commands

### DISCUSSION THREAD

**Discussion by dnewman-gpsw** [gopro/labs#1199] [Announcements]
This is very new and very experimental, it only works so-so.

Video demo for Labs guided GPT https://www.youtube.com/shorts/5lCHP6VPk0c
Direct link to the GPT (likely requires an OpenAI/ChatGPT account)  https://bit.ly/chat_gp_labs

All feedback welcome.

**Comment by RikshaDriver** [comment]
Thanks for this handy integration!

Tried a query:

_Write a boot script to turn on GPS, sync time then turn off GPS_


Got the following output:



Here is the script to turn on GPS, synchronize time, and then turn off GPS:

```
!Sg1!t!2Ng0
```

### Explanation:
1. `!S`: Start immediately after the camera powers on.
2. `g1`: Enable GPS to allow synchronization.
3. `!t`: Synchronize the camera's time using the GPS signal.
4. `!2N`: Pause briefly to ensure time sync completes before the next action.
5. `g0`: Disable GPS to conserve battery after synchronization. 

This ensures GPS is only used as needed for the time sync.


Your thoughts on the output @dnewman-gpsw?

**Reply by dnewman-gpsw** [reply]
It has no idea on what boot script is yet, or any of the extensions, so it made most of it up.  !t - the some much say that is not valid, yet it will happily hallucinate that. It is been trained on mode settings, and actions, like do this or that at particular times. I will slowly expand its knowledge, but just point it at the web docs is nowhere near enough. It got the actions of the video, took about 8 hours of work. AI is still incredibly not very smart.

**Reply by dnewman-gpsw** [reply]
Added some understanding for boot commands.

**Reply by dnewman-gpsw** [reply]
I beefed up its knowledge base.

Your input:  "Write a boot script to turn on GPS, sync time then turn off GPS"

New result:
Here is the requested GoPro Labs boot script that activates GPS, syncs the time, and then disables GPS:

`*BOOT="!Lbt"!SAVEbt=g1!Dg0`

Explanation:
`g1`: Turns on GPS.
`!D`: Syncs the time using GPS.
`g0`: Turns off GPS.
`*BOOT="!Lbt"`: Specifies that this script is a boot script and should load automatically when the camera starts.
`!SAVEbt=g1!Dg0`: Saves the commands into the bt file for boot execution.
This ensures the camera activates GPS, syncs time, and then disables GPS on every boot. 

----
This will work if you had *SYNC=1 previously set.

**Reply by RikshaDriver** [reply]
Always fun to see how the AI iterates... latest output of this query:


Here is a boot script to turn on GPS, synchronize the time using GPS, and then turn off GPS:

```
*BOOT="!Lbt"!SAVEbt=g1!D!2Ng0
```

### Explanation:
1. `*BOOT="!Lbt"`: Declares a boot script named "bt".
2. `g1`: Turns on GPS.
3. `!D`: Syncs the camera's clock using GPS time.
4. `!2N`: Adds a short delay to ensure GPS synchronization is complete.
5. `g0`: Turns off GPS.

This sequence ensures that the GPS is only active long enough to synchronize the time, saving battery life.
